### PR TITLE
add option FHEROES2_WITH_SYSTEM_SMACKER

### PR DESCRIPTION
**Pull request by undef21**
to use system libsmacker instead bundled one.

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @undef21 , can you please explain the reason of using the system library? As far as I remember we ourselves fixed some bugs in this library which comes with the code.

**Comment by undef21** [comment]
Hello, @ihhub, @oleg-derevenetz .

Many linux distributions have policies which require to use system libraries as much as possible instead bundled ones. So this PR is mainly for distribution maintainers.

> Yes, I upgraded it to the latest version some time ago, the original libsmacker (without our patches) was nearly unusable, so I removed the option to use the non-bundled one.

Now Debian and Ubuntu have `libsmacker` in their repositories ([Debian](https://packages.debian.org/search?suite=all&section=all&arch=any&searchon=names&keywords=libsmacker-dev) and [Ubuntu](https://packages.ubuntu.com/search?keywords=libsmacker-dev&searchon=names)) with `tree-size-check` patch and `fheroes2` in that distributions already [uses](https://packages.debian.org/sid/fheroes2) system libsmacker. So I suggest restore building with external libraries instead bundled ones.

File: Makefile [hunk 1]
```diff
@@ -25,6 +25,7 @@
 # FHEROES2_WITH_ASAN: build with UB Sanitizer and Address Sanitizer (small runtime overhead, incompatible with FHEROES2_WITH_TSAN)
 # FHEROES2_WITH_TSAN: build with UB Sanitizer and Thread Sanitizer (large runtime overhead, incompatible with FHEROES2_WITH_ASAN)
 # FHEROES2_WITH_IMAGE: build with SDL_image (requires libpng)
+# FHEROES2_WITH_SYSTEM_SMACKER: build with external libsmacker instead bundled one
```
Comment (oleg-derevenetz) [inline Makefile:28]:
```suggestion
# FHEROES2_WITH_SYSTEM_SMACKER: build with an external libsmacker instead of the bundled one
```

**Review by oleg-derevenetz — CHANGES_REQUESTED** [review]
Hi @undef21 looks good, the only recommendation is to slightly rephrase the comment. I suppose this has been tested on systems where there is a `libsmacker` from packages?

File: Makefile (hunk 1 above)
Comment (undef21) [reply Makefile]: Thanks. Fixed.

Yes, this has been tested on Debian and Ubuntu, also another Debian derivatives (Kali, Raspbian) use the patched library. Another systems should patch their libsmacker packages to use this options or use the bundled library.

**Review by oleg-derevenetz — APPROVED** [review]

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@undef21 , huge thanks for this change!

**Pull request merged**